MediaBrowser.Providers/Plugins/MusicBrainz/MusicBrainzAlbumProvider.cs

```diff
@@ -1,805 +1,262 @@
-                                }
-                        }
-                    }
-                    else
-                    {
-                        reader.Read();
-                    }
-                }
-
-                return Enumerable.Empty<ReleaseResult>();
+                result.Item.SetProviderId(MetadataProvider.MusicBrainzAlbum, releaseId);
             }
 
-            private static IEnumerable<ReleaseResult> ParseReleaseList(XmlReader reader)
+            if (!string.IsNullOrEmpty(releaseGroupId))
             {
-                reader.MoveToContent();
-                reader.Read();
-
-                // Loop through each element
-                while (!reader.EOF && reader.ReadState == ReadState.Interactive)
-                {
-                    if (reader.NodeType == XmlNodeType.Element)
-                    {
-                        switch (reader.Name)
-                        {
-                            case "release":
-                                {
-                                    if (reader.IsEmptyElement)
-                                    {
-                                        reader.Read();
-                                        continue;
-                                    }
-
-                                    var releaseId = reader.GetAttribute("id");
-
-                                    using var subReader = reader.ReadSubtree();
-                                    var release = ParseRelease(subReader, releaseId);
-                                    if (release != null)
-                                    {
-                                        yield return release;
-                                    }
-
-                                    break;
-                                }
-
-                            default:
-                                {
-                                    reader.Skip();
-                                    break;
-                                }
-                        }
-                    }
-                    else
-                    {
-                        reader.Read();
-                    }
-                }
+                result.Item.SetProviderId(MetadataProvider.MusicBrainzReleaseGroup, releaseGroupId);
             }
+        }
 
-            private static ReleaseResult ParseRelease(XmlReader reader, string releaseId)
-            {
-                var result = new ReleaseResult
-                {
-                    ReleaseId = releaseId
-                };
-
-                reader.MoveToContent();
-                reader.Read();
+        return result;
+    }
 
-                // http://stackoverflow.com/questions/2299632/why-does-xmlreader-skip-every-other-element-if-there-is-no-whitespace-separator
+    /// <inheritdoc />
+    public Task<HttpResponseMessage> GetImageResponse(string url, CancellationToken cancellationToken)
+    {
+        throw new NotImplementedException();
```

Comment:
Should it really throw?

Reply:
I think just returning would make more sense.
AFAIK implementing it is required by the interface.

What's your take?

Reply:
I'd rather we didn't throw exceptions. Is this the only provider that doesn't implement it?

Reply:
AudioDB also throws on some providers.

Reply:
Let's just keep it that way for now then